Libraries/Linking/NativeLinking.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ * @format
+ */
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+import Platform from 'Platform';
+
+export interface Spec extends TurboModule {
+  // Common interface
+  +getInitialURL: () => Promise<string>;
+  +canOpenURL: (url: string) => Promise<boolean>;
+  +openURL: (url: string) => Promise<void>;
+  +openSettings: () => Promise<void>;
+
+  // Android only
+  +sendIntent: (
+    action: string,
+    extras: ?Array<{key: string, value: string | number | boolean}>,
+  ) => Promise<void>;
+}
+
+const moduleName =
+  Platform.OS === 'ios' ? 'NativeLinkingManager' : 'NativeIntentAndroid';
+export default TurboModuleRegistry.getEnforcing<Spec>(moduleName);
```

Comment:
Let's not use a variable for module name, so that scripting can collect these callsites (we have an internal script that find all JS callsites for a TurboModule).

```
if (Platform.OS === 'ios') {
  export default TurboModuleRegistry.getEnforcing<Spec>('NativeLinkingManager');
} else {
  export default TurboModuleRegistry.getEnforcing<Spec>('NativeIntentAndroid');
}
```

@RSNara: we probably should rename these modules in the native side to have the same name...

Reply:
Cool

Reply:
@fkgozali btw, it is not legal to export that way.

Reply:
```

export default (Platform.OS === 'ios'
  ? TurboModuleRegistry.getEnforcing<Spec>('LinkingManager')
  : TurboModuleRegistry.getEnforcing<Spec>('IntentAndroid'));
```

^ is valid